FIAT/functional.py

```diff
@@ -71,39 +71,18 @@ def __init__(self, ref_el, target_shape,
         return
 
     def evaluate(self, f):
-        """Evaluates the functional on some callable object f."""
-        result = 0
-
-        # non-derivative part
```

Comment:
I'm pretty sure that what was there only ever worked for point evaluation. If you look at to_riesz, which is a close relative of __call__, it suggests that concrete functionals will want to subclass. I think that is the right call: each type of functional is pretty easy to evaluate but its a little complicated to write a general case.